mautic/mautic
# PHP 8.2 support

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR contains the minimal changes to get Mautic working on PHP 8.2 and get the CI green.

The UTF8 conversions aren't necessary since Mautic 3 when we started using UTF8MB4 encoding. I think it's safe to get rid of it. The methods used for it were removed in PHP 8.2.

The `ext-iconv` should be required as we use it in the code without any checks if it is installed or not.

##### Old description:

~~This PR is just to test if there is some BC break required to support PHP 8.2. Based on the PHPUNIT output there doesn't seem to be. Many deprecations and failed tests are due to Swiftmaier which is going to be removed in https://github.com/mautic/mautic/pull/11613. So we can do this any time after that is merged. Even after M5.0.0 release.~~

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Gitpod was updated to PHP 8.2 as well so you can test it on this newer PHP version. There is nothing specific to test. Everything should work as before.
3. One thing I'd recommend to focus on is sending emails with some UTF8 characters like emojis.
